src/scanner.cc

```diff
@@ -456,23 +456,7 @@ static function<void(State &)> consume_while(Peek pred) {
   };
 }
 
-// TODO this breaks if the target sequence has a repetition of its prefix
```

Comment:
Should we keep the comment? Or did you fix this?

Reply:
@tek can you advise? I think `mark_end` is fine to call multiple times, so I don't see why this would break if the needle contains its prefix more than once.